opengate/engines.py

```diff
@@ -499,6 +647,64 @@ def initialize_optical_material_properties(self):
                         f"found in volume {vol.name} from file {self.physics_manager.optical_properties_file}."
                     )
 
+    def initialize_surface_material_properties(self):
+        """
+
+        This function deals with calling other functions and storing the
+        information required to create optical surfaces.
+
+        This information stored is used to create an optical surface (G4LogicalBorderSurface).
+        """
+
+        volume_surfaces_info = (
+            self.simulation_engine.simulation.physics_manager.volume_surfaces_info
+        )
+
+        for key, surfaces in volume_surfaces_info.items():
+            for surface in surfaces:
+                volume_1 = surface["volumes"][0]
+                volume_2 = surface["volumes"][1]
+
+                # Calls the function to get the physical volumes
+                # required to create optical surface (G4LogicalBorderSurface)
+                self.g4_physical_volumes = get_g4_physical_volumes(volume_1, volume_2)
+
+                # Loads the surface properties of the specified surface from SurfaceProperties.xml
+                surface_properties = load_surface_properties_from_xml(
+                    self.physics_manager.surface_properties_file,
+                    surface["surface_name"],
+                )
+
+                # Creates a surface properties table with information from xml.
+                self.g4_surface_properties = create_g4_surface_properties(
+                    surface_properties, surface["surface_name"]
+                )
+
+                print("Entering this initialize_surface_material_properties block")
+
+                # Creates an Optical Surface with the surface finish specified by the user
+                self.g4_logical_surface = g4.G4LogicalBorderSurface(
+                    g4.G4String(surface["surface_name"]),
+                    self.g4_physical_volumes[0],
+                    self.g4_physical_volumes[1],
+                    self.g4_surface_properties,
+                )
+
+                print(f"The surface created is {self.g4_logical_surface}")
+
+                if self.g4_logical_surface is not None:
+                    self.g4_logical_surface_list.append(self.g4_logical_surface)
```

Comment:
I would prefer a dictionary self.g4_logical_surfaces rather than a list. The keys could be the same as in volume_surfaces_info (in that case, you do not to iterate over .items() - see above)